.github/workflows/delete-kots-channel.yml

```diff
@@ -0,0 +1,32 @@
+name: Branch Deleted
+on: delete
+env:
+  REPLICATED_CLI_VERSION: 0.40.1
+  REPLICATED_API_TOKEN: ${{ secrets.REPLICATED_API_TOKEN }}
+  REPLICATED_APP: ${{ secrets.REPLICATED_APP }}
+jobs:
+  delete:
+    if: github.event.ref_type == 'branch'
+    runs-on: ubuntu-latest
+    continue-on-error: true
+    steps:
+      - name: Install Replicated CLI
+        run: |
+          curl -sL https://github.com/replicatedhq/replicated/releases/download/v${{ env.REPLICATED_CLI_VERSION }}/replicated_${{ env.REPLICATED_CLI_VERSION }}_linux_amd64.tar.gz -o replicated.tar.gz
+          tar xf replicated.tar.gz replicated && rm replicated.tar.gz
+          mv replicated /usr/local/bin/replicated
+
+      - name: Delete Replicated channel
+        run: |
+          CHANNEL_ID=$(replicated channel inspect ${{ github.event.ref }} \
+            | grep "ID:" \
+            | sed "s/ID://" \
+            | sed "s/ //g" || true)
+
+          if [ "${CHANNEL_ID}" = "" ]; then
+            echo "No channel found"
+            exit 0
+          fi
+
+          # Allowed to fail if customers on channel - this will need to be manually deleted
+          replicated channel rm "${CHANNEL_ID}" || true
```

Comment:
TIOLI: How about we also do the check manually? Just want to be on the safer side, as its a v important operation.

Reply:
To the best of my knowledge, there's no way of checking if a channel has customers using the CLI. Honestly, it's all a little undocumented so I've had to dig through the [source code](https://github.com/replicatedhq/replicated/tree/main/cli/cmd). The closest we have is `channel counts` but keeps coming back with `Error: This feature is not supported for Kots applications.` which makes no sense.

It is documented that you cannot delete a channel with customers and there's no option to force delete. And we wouldn't habitually use branches with the channel names `stable`, `beta`, `unstable`, `enterprise`.

Reply:
Thanks for the elaborate answer. It makes sense.

Reply:
If there's any issues, we can always get revert this commit (I'm going to create/delete a branch now)